python/lsst/daf/butler/core/datastore.py

```diff
@@ -365,3 +372,27 @@ def transfer(self, inputDatastore, datasetRef):
             Reference to the required Dataset.
         """
         raise NotImplementedError("Must be implemented by subclass")
+
+    @abstractmethod
+    def validateConfiguration(self, *entities, logFailures=False):
+        """Validate some of the configuration for this datastore.
```

Comment:
What do you mean by "some of the configuration"? Parts of configuration related to entities? Or is it entirely up to a subclass?

Reply:
It's entirely up to the subclass and I'm only checking that templates and formatters are consistent. Even that is only done by PosixDatastore. InMemoryDatastore doesn't check anything at the moment.  The code isn't looking at every configuration item.